### PR TITLE
Generate binding redirects for tests that need it

### LINKED ISSUE #7620: Investigate failing tests that use Roslyn
The tests that compile code are failing on machines with both VS 2015 and VS 2017. They were disabled in 33bf2453f39f8328ef9f0c64ca4f547b0f2e4ee4, but we should investigate the issue and re-enable them. The issue may also go away when we stop using the 4.4.0 contracts.

### PR DESCRIPTION
**Pull request by bricelam**
Resolves #7620

### CODE REVIEW

**Comment by tinchou** [comment]
I just tried to run `SqlServerE2ETests` with TestDriven.NET and I got `PathTooLongException`. If I try to run `CSharpMigrationOperationGeneratorTest` everything works fine. If CI passes then I approve.

**Comment by bricelam** [comment]
@tinchou I hit that too. I'll see if there's a workaround...

**Comment by bricelam** [comment]
...if only we went with `EFCore` instead of `EntityFrameworkCore`... :trollface:

File: test/Microsoft.EntityFrameworkCore.Design.Tests/Microsoft.EntityFrameworkCore.Design.Tests.csproj [hunk 1]
```diff
@@ -4,6 +4,11 @@
     <TargetFrameworks>net452;netcoreapp1.1</TargetFrameworks>
     <TargetFrameworks Condition="'$(OS)' != 'Windows_NT'">netcoreapp1.1</TargetFrameworks>
     <PreserveCompilationContext>true</PreserveCompilationContext>
+    <!-- TODO: Remove when Microsoft/vstest#428 is fixed. -->
+    <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
+    <GenerateBindingRedirectsOutputType>true</GenerateBindingRedirectsOutputType>
+    <!-- TODO: Remove when dotnet/sdk#909 is fixed. -->
+    <OutputType>Exe</OutputType>
```
Comment (bricelam) [inline test/Microsoft.EntityFrameworkCore.Design.Tests/Microsoft.EntityFrameworkCore.Design.Tests.csproj]: Looks like this isn't needed.

**Comment by bricelam** [comment]
Blocked on #7779

**Pull request merged**